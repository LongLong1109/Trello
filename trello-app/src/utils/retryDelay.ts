export const retryDelay = (attemptIndex: number) => Math.min(1000 * 2 ** attemptIndex, 30000)
