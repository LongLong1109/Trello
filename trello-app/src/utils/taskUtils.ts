//interface
import { ListTask, TaskListItem, Task } from '@/interfaces/Task'
import { TASK_PROPERTY_KEY, TaskPropertyValue } from '@/types/Task'

export const updateTaskProperty = (
  lists: ListTask[],
  listId: number,
  taskId: number,
  key: TASK_PROPERTY_KEY,
  value: TaskPropertyValue,
) => {
  return lists.map((list) => {
    const isTargetList = list.id === listId
    if (isTargetList) {
      return {
        ...list,
        tasks: list.tasks.map((task) => (task.id === taskId ? { ...task, [key]: value } : task)),
      }
    }
    return list
  })
}

export const removeTaskById = (lists: ListTask[], listId: number, taskId: number) => {
  return lists.map((list) => {
    const isTargetList = list.id === listId
    if (isTargetList) {
      return {
        ...list,
        tasks: list.tasks.filter((task) => task.id !== taskId),
      }
    }
    return list
  })
}

export const updateTaskTitle = (lists: ListTask[], selectedTask: TaskListItem, title: string) => {
  return lists.map((list) =>
    list.id === selectedTask.listId
      ? {
          ...list,
          tasks: list.tasks.map((task) =>
            task.id === selectedTask.id ? { ...task, title } : task,
          ),
        }
      : list,
  )
}

export const moveTask = (
  taskToMove: Task,
  updatedLists: ListTask[],
  targetListId: number,
  targetIndex: number,
) => {
  return updatedLists.map((list) => {
    if (list.id === targetListId) {
      const updatedTasks = [...list.tasks]
      updatedTasks.splice(targetIndex, 0, taskToMove)
      return { ...list, tasks: updatedTasks }
    }
    return list
  })
}
