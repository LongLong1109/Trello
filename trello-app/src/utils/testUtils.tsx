import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { MantineProvider } from '@mantine/core'
const AllBrowserRouter = ({ children }: { children: React.ReactNode }) => {
  return <MantineProvider>{children}</MantineProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllBrowserRouter, ...options })

export * from '@testing-library/react'
export * from '@testing-library/user-event'
export { customRender as render }
