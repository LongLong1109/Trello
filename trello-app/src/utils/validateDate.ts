export const isDateOverdue = (date: Date) => {
  const currentDate = new Date()
  currentDate.setHours(0, 0, 0, 0)
  date.setHours(0, 0, 0, 0)

  const timeDifference = date.getTime() - currentDate.getTime()
  return timeDifference < 0
}

export const convertDateToString = (date: Date) => {
  const validDate = date ? new Date(date) : new Date()
  return validDate.toLocaleDateString()
}

export const formatDate = (date: Date): string => {
  if (!date || isNaN(date.getTime())) {
    return '---'
  }

  const options: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric' }
  return new Intl.DateTimeFormat('en-US', options).format(date)
}
