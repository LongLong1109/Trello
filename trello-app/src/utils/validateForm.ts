import { MESSAGES } from '@/constants/message'
import { REGEX_EMAIL, PASSWORD_MIN_LENGTH } from '@/constants/validateFields'

export const validateFirstName = (value: string) => {
  return value === '' ? MESSAGES.firstName : null
}

export const validateLastName = (value: string) => {
  return value === '' ? MESSAGES.lastName : null
}

export const validateEmail = (value: string) => {
  return REGEX_EMAIL.test(value) ? null : MESSAGES.email
}

export const validatePassword = (value: string) => {
  if (value === '') {
    return MESSAGES.passwordRequired
  }
  return value.length >= PASSWORD_MIN_LENGTH ? null : MESSAGES.password
}
