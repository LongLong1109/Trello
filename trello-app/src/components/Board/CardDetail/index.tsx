/* eslint-disable jsx-a11y/no-autofocus */
import { ChangeEvent } from 'react'
import { Flex, Paper, Title, Textarea, Modal, Badge, Skeleton } from '@mantine/core'
import { IconCalendar, IconTagStarred, IconMinus } from '@tabler/icons-react'

// utils
import { isDateOverdue } from '@/utils/validateDate'

// components
import { Labels, Button, DatePicker, Popover, CheckboxItem } from '../../common'

export interface CardDetailProps {
  labels: string[]
  description: string
  checkList: CheckboxItem[]
  dueDate: Date | null
  isOpenLabel: boolean
  isOpenDate: boolean
  formattedDate: string
  isEditingDescription: boolean
  comments: string
  isEditComment: boolean
  isOpenRemoveCard: boolean
  isLoading: boolean
  onEditComment: (value: boolean) => void
  onEditingDescription: (value: boolean) => void
  onUpdateComment: (value: string) => void
  onOpenLabel: (value: boolean) => void
  onOpenDate: (value: boolean) => void
  onUpdateDescription: (description: string) => void
  onUpdateLabels: (index: number, checked: boolean) => void
  onUpdateDueDate: (date: Date | null) => void
  onSaveDate: () => void
  onRemoveDate: () => void
  onOpenRemoveCard: (value: boolean) => void
  onRemoveCard: () => void
}

const CardDetail = ({
  labels,
  description,
  checkList,
  dueDate,
  isOpenLabel,
  isOpenDate,
  formattedDate,
  isEditingDescription,
  comments,
  isEditComment,
  isLoading,
  onEditComment,
  onUpdateComment,
  onEditingDescription,
  onOpenLabel,
  onOpenDate,
  onUpdateDescription,
  onUpdateLabels,
  onUpdateDueDate,
  onSaveDate,
  onRemoveDate,
  isOpenRemoveCard,
  onOpenRemoveCard,
  onRemoveCard,
}: CardDetailProps) => {
  const taskDueDate = new Date(dueDate ?? '')
  const checkDateOverdue = isDateOverdue(taskDueDate)

  const handleOpenLabel = () => {
    onOpenLabel(true)
  }

  const handleCloseLabel = () => {
    onOpenLabel(false)
  }

  const handleOpenDueDate = () => {
    onOpenDate(true)
  }

  const handleCloseDueDate = () => {
    onOpenDate(false)
  }

  const handleDescriptionUpdated = () => {
    onEditingDescription(true)
    onUpdateDescription(description)
  }

  const handleDescriptionChange = ({
    currentTarget: { value },
  }: ChangeEvent<HTMLTextAreaElement>) => {
    onUpdateDescription(value)
  }

  const handleDescriptionBlur = () => {
    onEditingDescription(false)
  }

  const handleCommentsUpdated = () => {
    onEditComment(true)
    onUpdateComment(comments)
  }

  const handleCommentsChange = ({ currentTarget: { value } }: ChangeEvent<HTMLTextAreaElement>) => {
    onUpdateComment(value)
  }

  const handleCommentsBlur = () => {
    onEditComment(false)
  }

  const handleOpenRemoveCard = () => {
    onOpenRemoveCard(true)
  }

  const handleCloseRemoveCard = () => {
    onOpenRemoveCard(false)
  }

  const handleRemoveCard = () => {
    onOpenRemoveCard(false)
  }

  return (
    <Flex justify='space-between' mih='500'>
      <Flex direction='column' w='60%'>
        <Flex direction='column' mb='20'>
          {isLoading ? (
            <Skeleton height={30} mb='10' />
          ) : (
            labels.length > 0 && (
              <Title c='backgrounds.8' order={6} mb='10'>
                Labels
              </Title>
            )
          )}
          <Flex gap='5'>
            {isLoading ? (
              <Skeleton width={50} height={30} />
            ) : (
              labels.map((label) => <Paper w='50' h='30' radius='4' bg={label} key={label} />)
            )}
          </Flex>

          {isLoading ? (
            <>
              <Skeleton height={30} mb='10' mt='10' />
              <Skeleton width={100} height={24} />
            </>
          ) : (
            dueDate && (
              <>
                <Title c='backgrounds.8' order={6} mb='10' mt='10'>
                  Due date
                </Title>
                <Badge radius='xs' color={checkDateOverdue ? 'red' : 'blue'}>
                  {taskDueDate.toLocaleDateString()} {checkDateOverdue && 'overdue'}
                </Badge>
              </>
            )
          )}
        </Flex>
        <Paper>
          <Title c='backgrounds.8' order={6} mb='10'>
            Description
          </Title>
          {isLoading ? (
            <Skeleton height={100} mb='sm' />
          ) : isEditingDescription ? (
            <Textarea
              value={description}
              onChange={handleDescriptionChange}
              onBlur={handleDescriptionBlur}
              autoFocus
              mb='sm'
              w='100%'
            />
          ) : (
            <Paper onClick={handleDescriptionUpdated} mb='sm' w='100%' p='10' bg='backgrounds.1'>
              {description || 'Add a more detailed description...'}
            </Paper>
          )}
        </Paper>

        <Paper>
          <Title c='backgrounds.8' order={6} mt='20' mb='10'>
            Comment
          </Title>
          {isLoading ? (
            <Skeleton height={100} mb='sm' />
          ) : isEditComment ? (
            <Textarea
              value={comments}
              onChange={handleCommentsChange}
              onBlur={handleCommentsBlur}
              autoFocus
              mb='sm'
              w='100%'
            />
          ) : (
            <Paper onClick={handleCommentsUpdated} mb='sm' w='100%' p='10' bg='backgrounds.4'>
              {comments || 'Write a comment...'}
            </Paper>
          )}
        </Paper>
      </Flex>

      <Flex direction='column' gap='10' w='30%'>
        <Title c='backgrounds.8' order={6}>
          Add to card
        </Title>
        <Popover
          name='Labels'
          icon={<IconTagStarred size='16' />}
          open={isOpenLabel}
          onOpen={handleOpenLabel}
          onClose={handleCloseLabel}
        >
          <Paper p='12' w='300'>
            {isLoading ? (
              <Skeleton height={40} />
            ) : (
              <Labels onChange={onUpdateLabels} checkList={checkList} onClose={handleCloseLabel} />
            )}
          </Paper>
        </Popover>

        <Popover
          name='Date'
          icon={<IconCalendar size='16' />}
          open={isOpenDate}
          onOpen={handleOpenDueDate}
          onClose={handleCloseDueDate}
        >
          <Paper p='12' w='300'>
            {isLoading ? (
              <Skeleton height={40} />
            ) : (
              <DatePicker
                value={dueDate}
                onChange={onUpdateDueDate}
                formattedDate={formattedDate}
                onSave={onSaveDate}
                onRemove={onRemoveDate}
              />
            )}
          </Paper>
        </Popover>

        <Modal.Root opened={isOpenRemoveCard} onClose={handleCloseRemoveCard} centered>
          <Modal.Overlay />
          <Modal.Content>
            <Modal.Header>
              <Flex align='center' justify='space-between' w='100%'>
                <Paper />
                <Modal.Title>Are you sure to remove this card</Modal.Title>
                <Modal.CloseButton m='0' />
              </Flex>
            </Modal.Header>
            <Modal.Body>
              <Flex p='12' justify='space-between'>
                <Button variant='light' onClick={handleRemoveCard} name='Cancel' />
                <Button color='red' onClick={onRemoveCard} name='Remove Card' />
              </Flex>
            </Modal.Body>
          </Modal.Content>
        </Modal.Root>
        <Button
          size='md'
          radius='sm'
          justify='flex-start'
          variant='light'
          leftSection={<IconMinus size='16' />}
          onClick={handleOpenRemoveCard}
          name='Remove Card'
        />
      </Flex>
    </Flex>
  )
}

export default CardDetail
