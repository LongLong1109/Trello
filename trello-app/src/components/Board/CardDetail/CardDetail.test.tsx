import CardDetail, { CardDetailProps } from '.';
import { render, screen, fireEvent } from '@/utils/testUtils';

const mockProps: CardDetailProps = {
  labels: ['blue', 'red'],
  description: 'Sample description',
  checkList: [{ label: 'Check item 1', checked: false, value: '', key: '1' }],
  dueDate: new Date('2024-05-27'),
  isOpenLabel: false,
  isOpenDate: false,
  formattedDate: 'May 27, 2024',
  isEditingDescription: false,
  comments: 'Sample comment',
  isEditComment: false,
  isOpenRemoveCard: false,
  onEditComment: jest.fn(),
  onEditingDescription: jest.fn(),
  onUpdateComment: jest.fn(),
  onOpenLabel: jest.fn(),
  onOpenDate: jest.fn(),
  onUpdateDescription: jest.fn(),
  onUpdateLabels: jest.fn(),
  onUpdateDueDate: jest.fn(),
  onSaveDate: jest.fn(),
  onRemoveDate: jest.fn(),
  onOpenRemoveCard: jest.fn(),
  onRemoveCard: jest.fn(),
};

const renderCardDetailProps = () => {
  return render(<CardDetail {...mockProps} />);
};

describe('<CardDetail />', () => {
  it('should be matches snapshot', () => {
    const { asFragment } = renderCardDetailProps();
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays labels and description correctly', () => {
    renderCardDetailProps();
    expect(screen.getByText('Sample description')).toBeInTheDocument();
  });

  it('allows editing description when clicked', () => {
    renderCardDetailProps();
    fireEvent.click(screen.getByText('Sample description'));
    expect(mockProps.onEditingDescription).toHaveBeenCalledWith(true);
  });

  it('toggles label popover', () => {
    renderCardDetailProps();
    const labelButton = screen.getAllByText('Labels');
    fireEvent.click(labelButton[1]);
    expect(mockProps.onOpenLabel).toHaveBeenCalledWith(true);
  });

  it('handles remove card action correctly', () => {
    renderCardDetailProps();
    const removeButton = screen.getByText('Remove Card');
    fireEvent.click(removeButton);
    expect(mockProps.onRemoveCard).toHaveBeenCalled();
  });
});
