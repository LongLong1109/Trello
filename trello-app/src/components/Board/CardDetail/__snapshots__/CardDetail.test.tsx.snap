// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDetail /> should be matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_8bffd616 mantine-Flex-root __m__-r0"
    style="justify-content: space-between; min-height: calc(31.25rem * var(--mantine-scale));"
  >
    <div
      class="m_8bffd616 mantine-Flex-root __m__-r2"
      style="flex-direction: column; width: 60%;"
    >
      <div
        class="m_8bffd616 mantine-Flex-root __m__-r4"
        style="flex-direction: column;"
      >
        <h1
          class="m_8a5d1357 mantine-Title-root"
          data-order="1"
          data-size="h2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Labels
        </h1>
        <div
          class="m_8bffd616 mantine-Flex-root __m__-r7"
          style="gap: calc(0.3125rem * var(--mantine-scale));"
        >
          <div
            class="m_1b7284a3 mantine-Paper-root"
            style="--paper-radius: calc(0.25rem * var(--mantine-scale));"
          />
          <div
            class="m_1b7284a3 mantine-Paper-root"
            style="--paper-radius: calc(0.25rem * var(--mantine-scale));"
          />
        </div>
        <h1
          class="m_8a5d1357 mantine-Title-root"
          data-order="1"
          data-size="h2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Due date
        </h1>
        <div
          class="m_347db0ec mantine-Badge-root"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            5/27/2024
          </span>
        </div>
      </div>
      <div
        class="m_1b7284a3 mantine-Paper-root"
      >
        <h1
          class="m_8a5d1357 mantine-Title-root"
          data-order="1"
          data-size="h2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Description
        </h1>
        <div
          class="m_1b7284a3 mantine-Paper-root"
          style="width: 100%;"
        >
          Sample description
        </div>
      </div>
      <div
        class="m_1b7284a3 mantine-Paper-root"
      >
        <h1
          class="m_8a5d1357 mantine-Title-root"
          data-order="1"
          data-size="h2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Comment
        </h1>
        <div
          class="m_1b7284a3 mantine-Paper-root"
          style="width: 100%;"
        >
          Sample comment
        </div>
      </div>
    </div>
    <div
      class="m_8bffd616 mantine-Flex-root __m__-rj"
      style="gap: calc(0.625rem * var(--mantine-scale)); flex-direction: column; width: 30%;"
    >
      <h1
        class="m_8a5d1357 mantine-Title-root"
        data-order="1"
        data-size="h2"
        style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
      >
        Add to card
      </h1>
      <button
        aria-controls="mantine-t2sphrkqx-dropdown"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-size="md"
        data-variant="light"
        data-with-left-section="true"
        id="mantine-t2sphrkqx-target"
        justify="flex-start"
        style="--button-justify: flex-start; --button-height: var(--button-height-md); --button-padding-x: var(--button-padding-x-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-sm); --button-bg: var(--mantine-color-blue-light); --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_a74036a mantine-Button-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-tag-starred "
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.5 7.5m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0"
              />
              <path
                d="M3 6v5.172a2 2 0 0 0 .586 1.414l7.71 7.71a2.41 2.41 0 0 0 3.408 0l5.592 -5.592a2.41 2.41 0 0 0 0 -3.408l-7.71 -7.71a2 2 0 0 0 -1.414 -.586h-5.172a3 3 0 0 0 -3 3z"
              />
              <path
                d="M12.5 13.847l-1.5 1.153l.532 -1.857l-1.532 -1.143h1.902l.598 -1.8l.598 1.8h1.902l-1.532 1.143l.532 1.857z"
              />
            </svg>
          </span>
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Labels
          </span>
        </span>
      </button>
      <button
        aria-controls="mantine-sh1ml9k69-dropdown"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-size="md"
        data-variant="light"
        data-with-left-section="true"
        id="mantine-sh1ml9k69-target"
        justify="flex-start"
        style="--button-justify: flex-start; --button-height: var(--button-height-md); --button-padding-x: var(--button-padding-x-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-sm); --button-bg: var(--mantine-color-blue-light); --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_a74036a mantine-Button-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-calendar "
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 7a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v12a2 2 0 0 1 -2 2h-12a2 2 0 0 1 -2 -2v-12z"
              />
              <path
                d="M16 3v4"
              />
              <path
                d="M8 3v4"
              />
              <path
                d="M4 11h16"
              />
              <path
                d="M11 15h1"
              />
              <path
                d="M12 15v3"
              />
            </svg>
          </span>
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Date
          </span>
        </span>
      </button>
      <button
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-size="md"
        data-variant="light"
        data-with-left-section="true"
        justify="flex-start"
        style="--button-justify: flex-start; --button-height: var(--button-height-md); --button-padding-x: var(--button-padding-x-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-sm); --button-bg: var(--mantine-color-blue-light); --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_a74036a mantine-Button-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-minus "
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12l14 0"
              />
            </svg>
          </span>
          <span
            class="m_811560b9 mantine-Button-label"
          >
            Remove Card
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
